torchreid/losses/asl.py

```diff
@@ -77,6 +76,17 @@ def __init__(self, m=0.35, k=0.8, t=1, s=30,
         # prevent memory allocation and gpu uploading every iteration, and encourages inplace operations
         self.anti_targets = self.xs_pos = self.xs_neg = self.asymmetric_w = self.loss = None
 
+    @staticmethod
+    def _valid(params):
+        if isinstance(params, (list, tuple)):
```

Comment:
Is aug_index a hierarchical structure?

Reply:
No, but we pass `logits_aug_avai = self._valid([aug_index, lam])`, a list of parameters and we iterate over this list to check each parameter